pkg/controller/awsmachine/awsmachine_controller.go

```diff
@@ -69,18 +103,283 @@ type ReconcileAWSMachine struct {
 // Reconcile reads that state of the cluster for a AWSMachine object and makes changes based on the state read
 // and what is in the AWSMachine.Spec
 func (r *ReconcileAWSMachine) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	// Fetch the AWSMachine instance
-	instance := &infrastructurev1alpha2.AWSMachine{}
-	err := r.Get(context.TODO(), request.NamespacedName, instance)
+	ctx := context.Background()
+
+	// Fetch the AWSMachine instance.
+	awsm := &infrav1.AWSMachine{}
+	err := r.Get(ctx, request.NamespacedName, awsm)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			// Object not found, return.  Created objects are automatically garbage collected.
-			// For additional cleanup logic use finalizers.
+		if apierrors.IsNotFound(err) {
 			return reconcile.Result{}, nil
 		}
-		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
 
+	// Store Machine early state to allow patching.
+	patchFrom := client.MergeFrom(awsm.DeepCopy())
```

Comment:
The scope's patch is a private field at the moment, I think this one is only used for adding finalizers, I might move it inside the finalizer block, wdyt?

Reply:
Even so, if you create the scope in advance of this, then the deferred `Close()` should do the patch.

Since we are patching now, we may not even need the early return anymore.

Reply:
Even better!